### PR TITLE
[AIRFLOW-3398] Google Cloud Spanner instance database query operator

### PR DESCRIPTION
**Pull request by sprzedwojski**
### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW-3398) issues and references them in the PR title.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Created an operator to enable executing arbitrary DML query (INSERT, UPDATE, DELETE) in a Transaction in Cloud Spanner:
`CloudSpannerInstanceDatabaseQueryOperator`

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

`test_gcp_spanner_operator.py`

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/contrib/example_dags/example_gcp_spanner.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+INSERT my_table2 (id, name) VALUES (7, 'Seven');
+INSERT
+my_table2 (id,
+name)
+VALUES (8,
+'Eight');
```
Comment (kaxil) [inline airflow/contrib/example_dags/example_gcp_spanner.sql]: Why is the 2nd query multiline?

File: airflow/contrib/operators/gcp_spanner_operator.py [hunk 2]
```diff
@@ -130,3 +132,67 @@ def execute(self, context):
...
+        if not self.query:
+            raise AirflowException("The required parameter 'query' is empty")
+
+    def execute(self, context):
+        queries = self.query
+        if isinstance(self.query, six.string_types):
+            queries = [x.strip() for x in self.query.split(';')]
+            self.sanitize_queries(queries)
+        self.log.info("Executing DML query(-ies) on projects/{}/instances/{}/databases/{}"
+                      .format(self.project_id, self.instance_id, self.database_id))
```
Comment (kaxil) [inline airflow/contrib/operators/gcp_spanner_operator.py]: Can we change this to %s format for logging  - https://docs.python.org/3/howto/logging-cookbook.html

Comment (kaxil) [reply airflow/contrib/operators/gcp_spanner_operator.py]: > Using logger.info("string template {}".format(argument)) should be avoided whenever possible in favor of logger.info("string template %s", argument). This is a better practice, as the actual string interpolation will be used only if the log will be emitted. Not doing so can lead to wasted cycles when we are logging on a level over INFO, as the interpolation will still occur.

File: airflow/contrib/example_dags/example_gcp_spanner.sql (hunk 1 above)
Comment (sprzedwojski) [reply airflow/contrib/example_dags/example_gcp_spanner.sql]: This is to show that we support multiline queries. This file is used in the example DAG (which can also be run as an integration test).
I changed it to 2 lines, as it looked a bit "weird" split up this way to be fair.

File: airflow/contrib/operators/gcp_spanner_operator.py (hunk 2 above)
Comment (sprzedwojski) [reply airflow/contrib/operators/gcp_spanner_operator.py]: Yes! Good point, I changed it, thanks.

**Review by kaxil — APPROVED** [review]

**Pull request merged**